tests/network/vmi_istio.go

```diff
@@ -117,6 +118,50 @@ var _ = SIGDescribe("[Serial] Istio", func() {
 			By("Waiting for VMI to be ready")
 			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
 		})
+		Describe("Live Migration", func() {
+			BeforeEach(func() {
+				tests.SkipIfMigrationIsNotPossible()
+			})
+			runMigrationAndExpectCompletion := func(migration *v1.VirtualMachineInstanceMigration, timeout int) {
+				migration, err = virtClient.VirtualMachineInstanceMigration(migration.Namespace).Create(migration)
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(func() error {
+					migration, err := virtClient.VirtualMachineInstanceMigration(migration.Namespace).Get(migration.Name, &metav1.GetOptions{})
+					if err != nil {
+						return err
+					}
+
+					Expect(migration.Status.Phase).ToNot(Equal(v1.MigrationFailed))
+
+					if migration.Status.Phase == v1.MigrationSucceeded {
+						return nil
+					}
+
+					return fmt.Errorf("migration is in phase %s", migration.Status.Phase)
+				}, timeout, time.Second).Should(Succeed(), fmt.Sprintf("migration should succeed after %d s", timeout))
+			}
+			expectAllContainersComplete := func(podName string) {
+				Eventually(func() error {
```

Comment:
I am not sure maybe the `Eventually` should be in the test itself, and this function should only get the pod and check the containers state.The thumb rule is try and avoid asserting in helpers.  I am not sure it is necessary in also in the second one on line 125, but still...  Then you can change the helper name to something like areAllContainersComplete.

```
areAllContainersComplete := func(podName string) (bool, error) {
		sourcePod, err := virtClient.CoreV1().Pods(vmi.Namespace).Get(context.TODO(), podName, metav1.GetOptions{})
		for _, containerStatus := range sourcePod.Status.ContainerStatuses {
			if containerStatus.State.Terminated == nil {
				return false, fmt.Errorf("container %s is not terminated, state: %s", containerStatus.Name, containerStatus.State.String())
			}
		}
		return true, nil
	}
```
something lke that at least...

Then you can even check if you want that the containers are not completed  before you send the quitquitquit,
and completely prove your point in this PR.

Reply:
Thanks, I have removed `Expect` and `Eventually` blocks from helper functions.

> Then you can even check if you want that the containers are not completed before you send the quitquitquit,
and completely prove your point in this PR.

I'm not sure I follow. Isn't adding the failing test before adding the call to `/quitquitquit` and then removing the `XFail` enough?

Reply:
@rhrazdil sorry for the late reply. I missed your comment here. 
Maybe it is enough, but providing a negative tests, do your change, and then run the positive test makes a stronger point I think. But as I see it it is not a must.